app/src/test/java/org/simple/clinic/registration/facility/RegistrationFacilitySelectionScreenControllerTest.kt

```diff
@@ -134,17 +138,46 @@ class RegistrationFacilitySelectionScreenControllerTest {
 
   @Test
   fun `when both facilities and location are fetched only then should facilities be shown`() {
-    // TODO
-  }
+    configProvider.onNext(configTemplate.copy(locationListenerExpiry = Duration.ofSeconds(5)))
 
-  @Test
-  fun `when facilities are fetched but location is still ongoing then facilities shouldn't be shown`() {
-    // TODO
+    val facilities = listOf(
+        PatientMocker.facility(name = "Facility 1"),
+        PatientMocker.facility(name = "Facility 2"))
+    whenever(facilityRepository.facilities(any())).thenReturn(Observable.just(facilities))
+    whenever(facilityRepository.recordCount()).thenReturn(Observable.just(facilities.size))
+    whenever(facilitySync.pullWithResult()).thenReturn(Single.just(FacilityPullResult.Success()))
+
+    whenever(locationRepository.streamUserLocation(any())).thenReturn(
+        Observable
```

Comment:
Instead of doing this, why don't we use a `PublishSubject<LocationUpdate>` instead?

Reply:
That is indeed more understandable. I'll do this.